.github/actions/set_branch_name/action.yml

```diff
@@ -0,0 +1,16 @@
+name: Set BRANCH_NAME environment variable
```

Comment:
nit: Do you think `BRANCH_NAME` might cause problems eventually since it's an environment variable made accessible to anything in the environment?

Reply:
> Do you think that might be premature?

Hmm yes maybe.

I have this thing bothering me, that we tend to end up having the same things reimplemented multiple times in disparate actions and I want to encourage people to reuse existing ones, but that said I don't think me introducing this will make anyone reuse it, they'll need to know it exists and as you imply, we don't know if anyone needs it yet anyway.

I think you're right, I'll bring this into the workflow instead. 😃